pkg/predicate/condition_status.go

```diff
@@ -22,8 +22,31 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
+// ConditionChangeFn is a type for comparing two conditions.
+type ConditionChangeFn func(con1, con2 *resourcesv1alpha1.ManagedResourceCondition) bool
+
+// DefaultConditionChange compares the given conditions and returns `true` if the `Status` has changed.
+var DefaultConditionChange ConditionChangeFn = func(con1, con2 *resourcesv1alpha1.ManagedResourceCondition) bool {
+	if con1 == nil {
+		// trigger if condition was added
+		return con2 != nil
+	}
+
+	if con2 == nil {
+		return true // condition was removed
+	}
+
+	return con1.Status != con2.Status
+}
+
+// ConditionChangedToUnhealthy compares the given conditions and returns if the `Status` has changed to an unhealthy state.
+var ConditionChangedToUnhealthy ConditionChangeFn = func(con1, con2 *resourcesv1alpha1.ManagedResourceCondition) bool {
+	return (con1 == nil || con1.Status == resourcesv1alpha1.ConditionTrue) &&
```

Comment:
Can you elaborate, why you want to trigger a reconciliation, if the condition was added (with Status `False`)?
I think, this can't happen anyways... So I guess, this doesn't matter 🤔 
But even if it could happen, an additional reconcile for newly created and broken MRs won't help probably...

Reply:
Yes, you're right that it's not very likely to happen. Eventually, I wanted it to be on the safe side and thought it wouldn't harm either to reconcile again.

Reply:
You're right. Thanks for clarifying :)